backend/lib/spree/backend/engine.rb

```diff
@@ -11,6 +11,12 @@ class Engine < ::Rails::Engine
       initializer 'spree.params.filter' do |app|
         app.config.filter_parameters += [:password, :password_confirmation, :number]
       end
+
+      initializer 'spree.core.checking_deprecated_preferences' do
+        Spree::Backend::Config.deprecated_preferences.each do |pref|
+          warn "[DEPRECATION] #{pref[:name]} is deprecated #{pref[:message]}"
```

Comment:
@Vegann what will be the value of the name? the preference key? if yes we need to point them to which gem preference is that, it should be `Spree::Backend::Config[:some_preference]`